backend/webapp/src/main/java/org/bonitasoft/web/designer/controller/upload/WebResourceUploader.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (C) 2015 Bonitasoft S.A.
+ * Bonitasoft, 32 rue Gustave Eiffel - 38000 Grenoble
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2.0 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.bonitasoft.web.designer.controller.upload;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+import org.bonitasoft.web.designer.controller.utils.MimeType;
+import org.bonitasoft.web.designer.model.WebResourceable;
+import org.bonitasoft.web.designer.model.resource.WebResource;
+import org.bonitasoft.web.designer.repository.Repository;
+import org.bonitasoft.web.designer.repository.WebResourceRepository;
+import org.bonitasoft.web.designer.repository.exception.NotFoundException;
+import org.bonitasoft.web.designer.rest.ErrorMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.multipart.MultipartFile;
+
+public class WebResourceUploader<T extends WebResourceable> {
+
+    protected static final Logger logger = LoggerFactory.getLogger(WebResourceUploader.class);
+
+    private Repository<T> repository;
+    private WebResourceRepository<T> webResourceRepository;
+
+    public WebResourceUploader(Repository<T> repository, WebResourceRepository<T> webResourceRepository) {
+        this.repository = repository;
+        this.webResourceRepository = webResourceRepository;
+    }
+
+    public ResponseEntity<ErrorMessage> upload(MultipartFile file, String id) {
+
+        if (file == null || file.isEmpty()) {
+            ErrorMessage errorMessage = new ErrorMessage("Argument", "Part named [file] is needed to successfully import a component");
+            logger.error(errorMessage.getMessage());
+            return new ResponseEntity<>(errorMessage, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        if (!MimeType.WEB_RESOURCES.matches(file.getContentType())) {
```

Comment:
MimeType is deleted. I use now an enum to define the type (STYLE, JAVASCRIPT or IMAGE)